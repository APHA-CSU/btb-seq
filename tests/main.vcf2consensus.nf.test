nextflow_process {

    name "Test Process vcf2Consensus"
    script "main.nf"
    process "vcf2Consensus"

    test("Should produce a masked consensus fasta file from input vcf and bed files") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = ["phix-m",
                            file("${projectDir}/tests/data/phix_test/mask.bed"),
                            file("${projectDir}/tests/data/phix_test/nonmasked-regions.bed"),
                            file("${projectDir}/tests/data/phix_test/phix-m.vcf.gz"), 
                            file("${projectDir}/tests/data/phix_test/phix-m.vcf.gz.csi"),
                            file("${projectDir}/tests/data/phix_test/phix-m.fas")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.consensus).match()
            assert path(process.out.nCount[0]).readLines().contains('Sample,Ncount,ResultLoc')
        }

    }

}
