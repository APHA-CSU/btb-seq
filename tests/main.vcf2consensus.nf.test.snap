{
    "Should produce a masked consensus fasta file from input vcf and bed files": {
        "content": [
            [
                [
                    "phix-m",
                    "phix-m_consensus.fas:md5,bd746f7f7cd7c614e7a3aa54dd79ff0d",
                    "phix-m_snps.tab:md5,0cd4e870c32929ae285754f3c1bf1bf9"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-03T23:04:14.705275033"
    }
}